### ISSUE TITLE
CheckWin() not working

### ISSUE THREAD

**Issue by bitcoin4cashqc** [ChipaDevTeam/BinaryOptionsToolsV1#29]
`True
Connecting...
WARNING:BinaryOptionsTools.platforms.pocketoption.ws.client:Error connecting, 'NoneType' object has no attribute 'join'
CONNECTED SUCCESSFUL
ERROR:root:Timeout: Could not retrieve order info in time.
(None, 'unknown')`

Sometimes it work, sometimes not. Why?

**Comment by Rick-29** [comment]
It's also since the api is sync it doesn't wait forever and waits up to 120 seconds (2mins) before failing if it doesn't recieve  a response for the server.

**Comment by Rick-29** [comment]
We will fix it for the v2 of the api

**Comment by bitcoin4cashqc** [comment]
> It's also since the api is sync it doesn't wait forever and waits up to 120 seconds (2mins) before failing if it doesn't recieve a response for the server.

But I wonder why there is no answers from the server.

**Comment by bitcoin4cashqc** [comment]
Do you have an idea when v2? I made a telegram/quotex bot that work very fine. I have modules for every groups I want (one of them use martingale so it's specific for that group only). I made a class for Pocket option using yours, it work, but if I can't get the result, it kinda defeat the purpose.

**Comment by theshadow76** [comment]
> Do you have an idea when v2? I made a telegram/quotex bot that work very fine. I have modules for every groups I want (one of them use martingale so it's specific for that group only). I made a class for Pocket option using yours, it work, but if I can't get the result, it kinda defeat the purpose.

I think it should be soon

**Comment by Rick-29** [comment]
> .

It's due to the fact that what the library does is check for a websocket message containing the close data and that is only recieved when the trade finishes so if the trade lasts 1 hour then if we hadn't the limit time, the thread would we stopped for 1 hour.

**Comment by Rick-29** [comment]
> Do you have an idea when v2? I made a telegram/quotex bot that work very fine. I have modules for every groups I want (one of them use martingale so it's specific for that group only). I made a class for Pocket option using yours, it work, but if I can't get the result, it kinda defeat the purpose.
I am still working on it, fixing some bugs and adding new features, then i will need to add wrappers for python. So a cupple of weeks I think.

**Comment by bitcoin4cashqc** [comment]
So basically if trade over 120 seconds it fail? I can edit that will try. Us that also mean I cant let's say get the trade status by id after closing amd reconnecting the websocket?

Obtenir Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Rick-29** [comment]
Yeah, you can change it, and it should work even after reconnecting. If the trade is longer than 120 seconds you could create a function to only call the check win and pass the trade id like 30 seconds before the trade ends

**Comment by theshadow76** [comment]
Fixed in [BinaryOptionsToolsv2](https://github.com/ChipaDevTeam/BinaryOptionsTools-v2)
